runHarness/ComputeResources/KubernetesCluster.pm

```diff
@@ -309,13 +309,46 @@ sub kubernetesDeleteAllWithLabel {
 	my $cmd;
 	my $outString;
 	my $cmdFailed;
-	$cmd = "kubectl delete all --selector=$selector --namespace=$namespace --kubeconfig=$kubeconfigFile $contextString";
+	$cmd = "kubectl delete deployments --selector=$selector --namespace=$namespace --kubeconfig=$kubeconfigFile $contextString";
 	($cmdFailed, $outString) = runCmd($cmd);
 	if ($cmdFailed) {
-		$logger->error("kubernetesDeleteAllWithLabel delete all failed: $cmdFailed");
+		$logger->error("kubernetesDeleteAllWithLabel delete deployments failed: $cmdFailed");
 	}
 	$logger->debug("Command: $cmd");
 	$logger->debug("Output: $outString");
+
+	$cmd = "kubectl delete sts --selector=$selector --namespace=$namespace --kubeconfig=$kubeconfigFile $contextString";
+	($cmdFailed, $outString) = runCmd($cmd);
+	if ($cmdFailed) {
+		$logger->error("kubernetesDeleteAllWithLabel delete sts failed: $cmdFailed");
+	}
+	$logger->debug("Command: $cmd");
+	$logger->debug("Output: $outString");
+
+	$cmd = "kubectl delete daemonsets --selector=$selector --namespace=$namespace --kubeconfig=$kubeconfigFile $contextString";
+	($cmdFailed, $outString) = runCmd($cmd);
+	if ($cmdFailed) {
+		$logger->error("kubernetesDeleteAllWithLabel delete daemonsets failed: $cmdFailed");
+	}
+	$logger->debug("Command: $cmd");
+	$logger->debug("Output: $outString");
+
+	$cmd = "kubectl delete replicasets --selector=$selector --namespace=$namespace --kubeconfig=$kubeconfigFile $contextString";
+	($cmdFailed, $outString) = runCmd($cmd);
+	if ($cmdFailed) {
+		$logger->error("kubernetesDeleteAllWithLabel delete replicasets failed: $cmdFailed");
+	}
+	$logger->debug("Command: $cmd");
+	$logger->debug("Output: $outString");
+
+	$cmd = "kubectl delete service --selector=$selector --namespace=$namespace --kubeconfig=$kubeconfigFile $contextString";
+	($cmdFailed, $outString) = runCmd($cmd);
+	if ($cmdFailed) {
+		$logger->error("kubernetesDeleteAllWithLabel delete service failed: $cmdFailed");
+	}
+	$logger->debug("Command: $cmd");
+	$logger->debug("Output: $outString");
+
 	$cmd = "kubectl delete configmap --selector=$selector --namespace=$namespace --kubeconfig=$kubeconfigFile $contextString";
```

Comment:
Doing it this way adds a lot of "No resources found" messages in the debug log due to not every selector/namespace having every type of resource to delete -- but that is ok.
Otherwise, seems to work without issues.

Reply:
I reduce the amount of logging by skipping the messages if the output is "No resource found".  There are still alot of runCmd log messages, but it is better.